xnat_tools/bids_postprocess.py

```diff
@@ -1,131 +1,84 @@
+    ),
+    includesubj: List[str] = typer.Option(
+        [],
+        "-i",
+        "--includeseq",
+        help="Include this participant only, can be specified multiple times",
+    ),
+    skipsubj: List[str] = typer.Option(
+        [],
+        "-s",
+        "--skipseq",
+        help="Skip this participant, can be specified multiple times",
+    ),
+    log_file: str = typer.Option("", help="File to senf logs to",),
+    verbose: bool = typer.Option(
+        False, "-v", help="Verbose logging. If True, sets loglevel to INFO"
+    ),
+    very_verbose: bool = typer.Option(
+        False, "--vv", help="Very verbose logging. If True, sets loglevel to DEBUG"
+    ),
+):
     """
-    parser = argparse.ArgumentParser(
-        description="Run Heudiconv on DICOMS form XNAT dump using Reproin Heuristic"
-    )
-    parser.add_argument(
-        "--session_suffix",
-        help="Suffix of the session for BIDS e.g, 01. This will produce a sesstion label of sess-01",
-        required=True,
-        type=str,
-    )
-    parser.add_argument(
-        "--bids_experiment_dir",
-        help="Root for BIDS for current experiment. The sub-* subfolders are expected under this directory",
-        required=True,
-    )
-    parser.add_argument(
-        "--subjlist",
-        help="List of participants to post process. If empty, all participants are processed. Accepts a list --subjlist 1 2 3",
-        required=False,
-        default=[],
-        nargs="*",  # 0 or more values expected => creates a list
-        type=str,
-    )
-    parser.add_argument(
-        "--skipsubj",
-        help="List of participants to SKIP. Accepts a list --skipsubj 1 2 3",
-        required=False,
-        default=[],
-        nargs="*",  # 0 or more values expected => creates a list
-        type=str,
-    )
-    parser.add_argument(
-        "-v",
-        "--verbose",
-        dest="loglevel",
-        help="set loglevel to INFO",
-        action="store_const",
-        const=logging.INFO,
-    )
-    parser.add_argument(
-        "-vv",
-        "--very-verbose",
-        dest="loglevel",
-        help="set loglevel to DEBUG",
-        action="store_const",
-        const=logging.DEBUG,
-    )
-
-    args, _ = parser.parse_known_args(args)
-    return args
-
-
-def setup_logging(loglevel):
-    """Setup basic logging
-
-    Args:
-      loglevel (int): minimum loglevel for emitting messages
+        Script for performing post BIDSIFY processing.
+        At the moment it inserts the IntendedFor field
+        to JSON sidecart for fieldmap data
     """
-    if loglevel is None:
-        loglevel = logging.INFO
 
-    logformat = "[%(asctime)s] %(levelname)s:%(name)s:%(message)s"
-    logging.basicConfig(
-        level=loglevel,
-        format=logformat,
-        datefmt="%Y-%m-%d %H:%M:%S",
-        handlers=[logging.StreamHandler(sys.stdout)],
+    setup_logging(
+        _logger, log_file, verbose=verbose, very_verbose=very_verbose,
```

Comment:
using the count scheme, if `verbose` is set to `3` we can send that integer into the `setup_logging` function instead of these two arguments. Then we pick log level based on the integer passed